### PR TITLE
Added solidity language server setup guide

### PR DESCRIPTION
**Pull request by kozlovzxc**
### Description

Just added a quick guide how to make LSP package to work with solidity, I've also tried to make [solc](https://docs.soliditylang.org/en/v0.8.23/installing-solidity.html#macos-packages) lsp to work, but [hardhat-vscode](https://github.com/NomicFoundation/hardhat-vscode/tree/development/server) feels much more suitable and easier to set up.

### Demo

image

image

### CODE REVIEW

File: docs/src/language_servers.md [hunk 1]
```diff
@@ -627,6 +627,23 @@ Follow installation instructions on [LSP-metals](https://github.com/scalameta/me
...
+
+1. Install [github:NomicFoundation/hardhat-vscode](https://github.com/NomicFoundation/hardhat-vscode/tree/development/server) (follow instructions in the repo):
+
+2. Open `Preferences > Package Settings > LSP > Settings` and add the `"solidity"` client configuration to the `"clients"`:
+
+    ```jsonc
+    {
+        "clients": {
+            "solidity": {
+                "command": ["/PATH/TO/node", "PATH/TO/nomicfoundation-solidity-language-server", "--stdio"],
```
Comment (rchl) [inline docs/src/language_servers.md:640]: Haven't checked but should be enough with something like:

```suggestion
                "command": ["nomicfoundation-solidity-language-server", "--stdio"],
```

File: docs/src/language_servers.md [hunk 2]
```diff
@@ -627,6 +627,23 @@ Follow installation instructions on [LSP-metals](https://github.com/scalameta/me
     }
     ```
 
+### Solidity
+
+1. Install [github:NomicFoundation/hardhat-vscode](https://github.com/NomicFoundation/hardhat-vscode/tree/development/server) (follow instructions in the repo):
+
+2. Open `Preferences > Package Settings > LSP > Settings` and add the `"solidity"` client configuration to the `"clients"`:
```
Comment (rchl) [inline docs/src/language_servers.md:634]: I'm guessing the user also has install some syntax for "solidity"?

File: docs/src/language_servers.md [hunk 3]
```diff
@@ -627,6 +627,23 @@ Follow installation instructions on [LSP-metals](https://github.com/scalameta/me
     }
     ```
 
+### Solidity
```
Comment (rchl) [inline docs/src/language_servers.md:630]: This is under the "Shell" programming language section. Use 2 hashes if it's a separate programming language.

File: docs/src/language_servers.md (hunk 1 above)
Comment (kozlovzxc) [reply docs/src/language_servers.md:640]: I've started with just `nomicfoundation-solidity-language-server`, but it didn't work for me, it wasn't able to locate `node` for some reason 🤔 Will take another look if there were any misconfig on my side

File: docs/src/language_servers.md (hunk 3 above)
Comment (kozlovzxc) [reply docs/src/language_servers.md:630]: Oh, my bad, will fix 👍

File: docs/src/language_servers.md (hunk 2 above)
Comment (jwortmann) [reply docs/src/language_servers.md:634]: The `source.solidity` scope used for `"selector"` is not built-in from Sublime Text, so you would need to have the according syntax package from Package Control installed to make the language server start. The other config examples on this page link to the syntax packages.

File: docs/src/language_servers.md (hunk 1 above)
Comment (rchl) [reply docs/src/language_servers.md:640]: If node is not in the path then I would consider it a separate issue that is explained at https://lsp.sublimetext.io/troubleshooting/#updating-the-path-used-by-lsp-servers

**Review by predragnikolic — APPROVED** [review]
Thanks! 
and sorry for the wait.

**Pull request merged**